### PR TITLE
Refactored bash service disable remediation + support for disable sockets

### PR DESCRIPTION
**Pull request by matejak**
The remediation got split into multiple functions.
The systemd service disable function now disables service and checks for availability of socket unit file. If it exists, it also disables socket.
I have checked usage of the shared remediation function and ensured that the refactoring doesn't break current use cases.

### CODE REVIEW

**Comment by mpreisler** [comment]
Maybe it's just me but could you please declare functions before you use them? It makes it MUCH more readable to people unfamiliar with your code / coding style. From multiple reviews of your code I suspect you like writing a high-level function and then filling in the functions doing the heavy lifting. That is fine but please reorganize it so that people reading your code can read it top to bottom.

I also think we should just use jinja2 to do the heavy-lifting here. However that's not in the scope of this PR so I won't insist on it.

**Comment by matejak** [comment]
For reference, the ordering here [has a reason](https://softwareengineering.stackexchange.com/a/138325).
However, we have agreed that the ordering of functions in the project is already mainly the other way (low-level first, highest level last).

**Comment by redhatrises** [comment]
So.... the whole point of the `service_command` was so that we wouldn't have to go to this level in our bash remediation scripts, but I guess that having just `service_command` was too confusing?

**Comment by mpreisler** [comment]
> So.... the whole point of the service_command was so that we wouldn't have to go to this level in our bash remediation scripts, but I guess that having just service_command was too confusing?

I agree that this PR goes a bit too far, especially considering this is going to get pasted all over the place in the XCCDF. Great bash style has its place but for remediations we probably want as compact as possible. In the future we should also just use jinja2 and avoid as much runtime detection as we can.

**Comment by matejak** [comment]
Well, if jinja is used on top of this PR, then the actual remediation code will be short and in good style. Moreover, pasting of helper functions all over again is a build system quirk that can be fixed at the source.
I vote very strongly for keeping the functionality extracted, as it will be easy to convert into macros, that will produce short code (good for users) from more verbose source (good for us devs).

File: shared/bash_remediation_functions/service_command.sh [hunk 1]
```diff
@@ -3,72 +3,81 @@
...
-
-  if [ "$service_operation" = 'disable' ] ; then
-    sed -i "s/disable.*/disable         = no/gI" /etc/xinetd.d/$xinetd
-  else
-    sed -i "s/disable.*/disable         = yes/gI" /etc/xinetd.d/$xinetd
-  fi
-fi
+	# If systemctl is installed, use systemctl command; otherwise, use the service/chkconfig commands
+	# foo_$service_state => foo_disabled / foo_enabled
+	if test -f "/usr/bin/systemctl"; then
```
Comment (mpreisler) [inline shared/bash_remediation_functions/service_command.sh]: you declared all the fancy variables and then use a hardcoded string here :-) I also think `which` would be more prudent.

**Comment by mpreisler** [comment]
I am sorry but I think this goes way too far. This is trivial code spanning a zillion lines and being extracted to multiple functions. IMO that's ridiculous...

**Comment by matejak** [comment]
Succeeded by #3055

**Pull request closed without merging**